mapie/tests/test_conformity_scores.py

```diff
@@ -0,0 +1,161 @@
+    "y_pred", [np.array([4, 7, 10, 12, 13, 12]), [4, 7, 10, 12, 13, 12]]
+)
+@pytest.mark.parametrize(
+    "conf_scores", [np.array([1, 0, -1, -1, 0, 3]), [1, 0, -1, -1, 0, 3]]
+)
+def test_absolute_conformity_score_get_estimation_distribution(
+    y_pred: ArrayLike, conf_scores: ArrayLike
+) -> None:
+    """Test conformity observed value computation for AbsoluteConformityScore."""  # noqa: E501
+    abs_conf_score = AbsoluteConformityScore()
+    y_obs = abs_conf_score.get_estimation_distribution(y_pred, conf_scores)
+    np.testing.assert_allclose(y_obs, y_toy)
+
+
+@pytest.mark.parametrize(
+    "y_pred", [np.array([4, 7, 10, 12, 13, 12]), [4, 7, 10, 12, 13, 12]]
+)
+def test_absolute_conformity_score_consistency(y_pred: ArrayLike) -> None:
+    """Test methods consistency for AbsoluteConformityScore."""
+    abs_conf_score = AbsoluteConformityScore()
+    signed_conf_scores = abs_conf_score.get_signed_conformity_scores(
+        y_toy, y_pred
+    )
+    y_obs = abs_conf_score.get_estimation_distribution(
+        y_pred, signed_conf_scores
+    )
+    np.testing.assert_allclose(y_obs, y_toy)
+
+
+@pytest.mark.parametrize(
+    "y_pred", [np.array([4, 7, 10, 12, 13, 12]), [4, 7, 10, 12, 13, 12]]
+)
+def test_gamma_conformity_score_get_conformity_scores(
+    y_pred: ArrayLike,
+) -> None:
+    """Test conformity score computation for GammaConformityScore."""
+    gamma_conf_score = GammaConformityScore()
+    conf_scores = gamma_conf_score.get_conformity_scores(y_toy, y_pred)
+    expected_signed_conf_scores = np.array(
+        [1 / 4, 0, -1 / 10, -1 / 12, 0, 3 / 12]
+    )
+    np.testing.assert_allclose(conf_scores, expected_signed_conf_scores)
+
+
+@pytest.mark.parametrize(
+    "y_pred", [np.array([4, 7, 10, 12, 13, 12]), [4, 7, 10, 12, 13, 12]]
+)
+@pytest.mark.parametrize(
+    "conf_scores",
+    [
+        np.array([1 / 4, 0, -1 / 10, -1 / 12, 0, 3 / 12]),
+        [1 / 4, 0, -1 / 10, -1 / 12, 0, 3 / 12],
+    ],
+)
+def test_gamma_conformity_score_get_estimation_distribution(
+    y_pred: ArrayLike, conf_scores: ArrayLike
+) -> None:
+    """Test conformity observed value computation for GammaConformityScore."""  # noqa: E501
+    gamma_conf_score = GammaConformityScore()
+    y_obs = gamma_conf_score.get_estimation_distribution(y_pred, conf_scores)
+    np.testing.assert_allclose(y_obs, y_toy)
+
+
+@pytest.mark.parametrize(
+    "y_pred", [np.array([4, 7, 10, 12, 13, 12]), [4, 7, 10, 12, 13, 12]]
+)
+def test_gamma_conformity_score_consistency(y_pred: ArrayLike) -> None:
+    """Test methods consistency for GammaConformityScore."""
+    gamma_conf_score = GammaConformityScore()
+    signed_conf_scores = gamma_conf_score.get_signed_conformity_scores(
+        y_toy, y_pred
+    )
+    y_obs = gamma_conf_score.get_estimation_distribution(
+        y_pred, signed_conf_scores
+    )
+    np.testing.assert_allclose(y_obs, y_toy)
+
+
+@pytest.mark.parametrize(
+    "y_pred", [np.array([4, 7, 10, 12, 13, 12]), [4, 7, 10, 12, 13, 12]]
```

Comment:
Include the lists in dedicated variables ?

Reply:
Done